### DISCUSSION TITLE
$GITHUB_PATH not working on macOS or Windows runners

### DISCUSSION THREAD

**Discussion by lorennorman** [community/community#80916] [Actions] [answered]
### Select Topic Area

Bug

### Body

Hey folks! 👋

I'm having a hard time updating the path on my non-Linux runners, I hope someone can help!

The [documentation says](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path) to use the `>> $GITHUB_PATH` style, and does not differentiate between runner OSes, which leads me to believe that the same mechanism should work on all runners.

(**Note:** one important caveat is that we set `shell: bash`, particularly on Windows runners which default to `powershell`.)

I've created a reproduction repository to demonstrate how the same action works fine on Linux, but fails on macOS and Windows. Please take a look:
https://github.com/lorennorman/mac-actions-bug

Am I doing something wrong? Am I missing some documentation somewhere?

Any help is appreciated!

**Comment by lorennorman** [comment, marked as answer]
Figured this out, you have to wait for the next step for the $PATH to be modified. Ubuntu was working because it already had the path I was adding on the `$PATH`! Here's [the fixed demonstration action](https://github.com/lorennorman/mac-actions-bug/actions/runs/7266804647) for anyone finding this later.

Closing.

**Comment by queenofcorgis** [comment]
We’re glad you could get this solved and we want to make sure others can do the same so we’ve reopened this discussion for better discoverability for our members.

**Comment by lorennorman** [comment]
Hey @queenofcorgis! Thanks for re-opening this for posterity, I didn't know how that worked.

I've been thinking 🤔 more about what was confusing here, and I have some feedback for you and/or the documentation team.

Looking at [the documentation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path) for the `$GITHUB_PATH` feature, it says:

> the currently running action cannot access the updated path variable.

...and I missed this, repeatedly. Maybe the example could cover it?

Next, the example demonstrates adding a path _that is already present_ on Ubuntu runners (likely the most common runner, by far.) This was a bit misleading, at least to me. This particular path (`"$HOME/.local/bin"`) is mentioned in many other places, as well.

My suggestions would be:
- expand the example to show setting the env var in one step, then accessing it in the next step, possibly including a bad access in the same step with a comment (`# THIS WON'T WORK!`)
- update how that particular path is used and discussed throughout the docs, a few ideas:
  - have it already in the $PATH for **all** runners, not just Ubuntu
  - explain it better where it's referenced, mention that it's already in Ubuntu but not others, etc
  - use a different path altogether for examples

Thanks for listening! Please reach out if you'd like any clarifications or more feedback! And thank you and the team for Actions, they're saving so much time for myself and my various orgs!

**Reply by queenofcorgis** [reply]
Hi there @lorennorman - you're actually able to make suggestions to our docs! You can open a pull request by selecting the make a contribution button at the bottom of the doc you'd like to update. Feel free to tag me and @Akash1134 on the PR as well for visibility.
